test/tests/unit/adopted-probes.test.ts

```diff
@@ -0,0 +1,155 @@
+		fetchSocketsStub.resolves([]);
+
+		adoptedProbes.scheduleSync();
+		await sandbox.clock.tickAsync(5000);
+
+		expect(whereStub.callCount).to.equal(1);
+		expect(whereStub.args[0]).to.deep.equal([{ ip: '1.1.1.1' }]);
+		expect(updateStub.callCount).to.equal(0);
+		expect(deleteStub.callCount).to.equal(1);
+		expect(deleteStub.callCount).to.equal(1);
+	});
+
+	it('class should do nothing if probe data is actual', async () => {
+		const adoptedProbes = new AdoptedProbes(sqlStub as unknown as Knex, fetchSocketsStub);
+		selectStub.resolves([{ ip: '1.1.1.1', uuid: '1-1-1-1-1', lastSyncDate: '1970-01-01' }]);
+		fetchSocketsStub.resolves([{ data: { probe: { ipAddress: '1.1.1.1', uuid: '1-1-1-1-1' } } }]);
+
+		adoptedProbes.scheduleSync();
+		await sandbox.clock.tickAsync(5000);
+
+		expect(whereStub.callCount).to.equal(0);
+		expect(updateStub.callCount).to.equal(0);
+	});
+
+	it('class update lastSyncDate if probe is connected and lastSyncDate < 30 days away', async () => {
+		const adoptedProbes = new AdoptedProbes(sqlStub as unknown as Knex, fetchSocketsStub);
+		selectStub.resolves([{ ip: '1.1.1.1', uuid: '1-1-1-1-1', lastSyncDate: '1969-12-31' }]);
+		fetchSocketsStub.resolves([{ data: { probe: { ipAddress: '1.1.1.1', uuid: '1-1-1-1-1' } } }]);
+
+		adoptedProbes.scheduleSync();
+		await sandbox.clock.tickAsync(5000);
+
+		expect(whereStub.callCount).to.equal(1);
+		expect(whereStub.args[0]).to.deep.equal([{ ip: '1.1.1.1' }]);
+		expect(updateStub.callCount).to.equal(1);
+		expect(updateStub.args[0]).to.deep.equal([{ lastSyncDate: new Date() }]);
+		expect(deleteStub.callCount).to.equal(0);
+	});
+
+	it('class update lastSyncDate if probe is connected and lastSyncDate > 30 days away', async () => {
+		const adoptedProbes = new AdoptedProbes(sqlStub as unknown as Knex, fetchSocketsStub);
+		selectStub.resolves([{ ip: '1.1.1.1', uuid: '1-1-1-1-1', lastSyncDate: '1969-11-15' }]);
+		fetchSocketsStub.resolves([{ data: { probe: { ipAddress: '1.1.1.1', uuid: '1-1-1-1-1' } } }]);
+
+		adoptedProbes.scheduleSync();
+		await sandbox.clock.tickAsync(5000);
+
+		expect(whereStub.callCount).to.equal(1);
+		expect(whereStub.args[0]).to.deep.equal([{ ip: '1.1.1.1' }]);
+		expect(updateStub.callCount).to.equal(1);
+		expect(updateStub.args[0]).to.deep.equal([{ lastSyncDate: new Date() }]);
+		expect(deleteStub.callCount).to.equal(0);
+	});
+
+	it('class update lastSyncDate should not update anything if lastSyncDate is today', async () => {
+		const adoptedProbes = new AdoptedProbes(sqlStub as unknown as Knex, fetchSocketsStub);
+		selectStub.resolves([{ ip: '1.1.1.1', uuid: '1-1-1-1-1', lastSyncDate: '1970-01-01' }]);
+		fetchSocketsStub.resolves([{ data: { probe: { ipAddress: '1.1.1.1', uuid: '1-1-1-1-1' } } }]);
+
+		adoptedProbes.scheduleSync();
+		await sandbox.clock.tickAsync(5000);
+
+		expect(whereStub.callCount).to.equal(0);
+		expect(updateStub.callCount).to.equal(0);
+		expect(deleteStub.callCount).to.equal(0);
+	});
+});
```

Comment:
No point in rewriting it now but note that here I'd greatly prefer integration tests that prepare data in a real DB, run the sync, and check the results. Would be much cleaner than all the mocking and provide better coverage as well.